include/boost/geometry/policies/relate/intersection_points.hpp

```diff
@@ -60,12 +61,20 @@ struct segments_intersection_points
         // denominator. In case of integer this results in an integer
         // division.
         BOOST_ASSERT(ratio.denominator() != 0);
+
+        typedef typename promote_integral<coordinate_type>::type promoted_type;
+
+        promoted_type numerator
+            = boost::numeric_cast<promoted_type>(ratio.numerator());
+        promoted_type denominator
+            = boost::numeric_cast<promoted_type>(ratio.denominator());
+        promoted_type dx_promoted = boost::numeric_cast<promoted_type>(dx);
+        promoted_type dy_promoted = boost::numeric_cast<promoted_type>(dy);
+
         set<0>(point, boost::numeric_cast<coordinate_type>(
-                get<0, 0>(segment)
-                    + ratio.numerator() * dx / ratio.denominator()));
+                get<0, 0>(segment) + numerator * dx_promoted / denominator));
         set<1>(point, boost::numeric_cast<coordinate_type>(
-                get<0, 1>(segment)
-                    + ratio.numerator() * dy / ratio.denominator()));
+                get<0, 1>(segment) + numerator * dy_promoted / denominator));
```

Comment:
Shouldn't call to `get()` be outside the outer `numeric_cast`? It may be not possible to add a value returned by this function to the object of promoted_type. So it should probably also be promoted explicitly or moved outside.

If you care about the overflow, it doesn't change anything since the result is converted to the `coordinate_type` in `set()` so there would be an overflow anyway. Or would numeric_cast throw an exception? If yes, would it be desireable?